test/statusline.cpp

```diff
@@ -0,0 +1,202 @@
+#include "statusline.h"
+
+#include "3rd-party/catch.hpp"
+
+using namespace newsboat;
+
+class StatusTester : public IStatus {
+public:
+	StatusTester()
+		: last_string_was_error(false)
+	{
+	}
+	void set_status(const std::string& message) override
+	{
+		last_message = message;
+		last_string_was_error = false;
+	}
+	void show_error(const std::string& message) override
+	{
+		last_message = message;
+		last_string_was_error = true;
+	}
+
+	std::string last_message;
+	bool last_string_was_error;
+};
+
+TEST_CASE("StatusLine passes status messages to registered IStatus implementation")
```

Comment:
Please add a second argument, "[StatusLine]", to this and the next test cases. That way, you can run *just* them with `./test/test '[StatusLine]'`, which can be helpful during debugging.